src/components/liveSession/components/liveSessionBtn/liveSessionBtn.component.js

```diff
@@ -98,29 +101,82 @@
                         };
 
                         return ZnkLessonNotesSrv.getLessonsByStudentIds([this.student.uid], dateRange, this.educatorProfile.uid)
                             .then(lessons => {
-                                return lessons.data && lessons.data.length ? lessons.data[0] : null;
-                            }, err => $log.debug('getScheduledLesson: getLessonsByStudentIds Error: ', err));
+                                return this.getLessonInRange(lessons.data);
+                            }, err => $log.error('getScheduledLesson: getLessonsByStudentIds Error: ', err));
                     });
 
 
                 };
+
+                this.getLessonInRange = (lessons) => {
+                    let scheduledLesson = null;
+                    if (lessons && lessons.length > 0) {
+                        if (lessons.length === 1) {
+                            scheduledLesson = lessons.pop();
+                        } else {
+                            $log.debug(`getLessonInRange: multiple lesson - check if it's back to back`);
+                            scheduledLesson = this.checkBack2BackLesson(lessons);
+                        }
+                    }
+                    $log.debug(`getLessonInRange: lessons is not defined or lessons.length === 0 `);
+                    return scheduledLesson;
+                };
+
+                this.checkBack2BackLesson = (lessons) => {
+                    $log.debug(`checkBack2BackLesson`);
+                    let scheduledLesson = null;
+                    let back2BackLessons = [];
+                    let now = Date.now();
+                    lessons.forEach(lesson => {
+                        let startTimeRange = lesson.date - SESSION_DURATION.marginBeforeSessionStart;
+                        let endTimeRange = lesson.date + SESSION_DURATION.length + SESSION_DURATION.marginAfterSessionStart;
+                        if ((startTimeRange < now < endTimeRange) && !scheduledLesson) {
+                            scheduledLesson = lesson;
+                            back2BackLessons.push(lesson);
+                        } else if (scheduledLesson) {
+                            if (back2BackLessons.length) {
+                                if ((back2BackLessons[back2BackLessons.length-1].date + SESSION_DURATION.length) === lesson.date) {
+                                    $log.debug(`checkBack2BackLesson: b2b lesson found. lessonId: ${lesson.id}`);
+                                    back2BackLessons.push(lesson);
+                                } else {
+                                    $log.debug(`checkBack2BackLesson: this lesson isn't b2b lesson.`);
+                                }
+                            }
+                        }
+                    });
+
+                    if (back2BackLessons.length > 1) {
+                        let newB2BLessonId = UtilitySrv.general.createGuid();
```

Comment:
do back2BackLessons.forEach
if one of them has already a backToBackId then do not create a new one, use the one you found instead.